File: Data Structures & Algorithms/non-overlapping-intervals/submission-0.cpp
class Solution {
public:

    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        
        int ans=0;

        int sp,ep;

        sort(intervals.begin(),intervals.end());
        sp= intervals[0][1];
        ep= intervals[0][1];

            
        int i=1;
        int x,y;
        while(i<intervals.size()){
            x = intervals[i][0];
            y = intervals[i][1];

            // cout<< sp << ' '<< ep<< ' '<< x << ' '<< y<< '\n';

            if(x < ep){
                if(ep > y){
                    sp = x;
                    ep = y;
                }
                ans++;
            }else {
                sp = x;
                ep = y;
            }
            i++;
        }

        return ans;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-0.java
class Solution {

    public boolean binarySearchOnAnswer(int []piles,int k,int h){
        System.out.println("next");
        for(int i=0;i<piles.length;i++){
            h -= ((piles[i]/k) + ((piles[i]%k == 0) ? 0 : 1));

            System.out.println(h + " "+ k + " "+ i);
            if(h<0)
                return false;
        }

       
        return true;

    }
    public int minEatingSpeed(int[] piles, int h) {
        int left =1;
        int right = piles[0];


        for(int i=0;i<piles.length;i++)
            right = Math.max(right,piles[i]);

        int mid;
        int ans=right;
        while(left<=right){
            mid = right - (right-left)/2;

            if(binarySearchOnAnswer(piles,mid,h)){
                ans = mid ;
                right = mid -1;
            }else left = mid+1;    

        }

        return ans;    
    }
}
